tests/test_core.py

```diff
@@ -0,0 +1,217 @@
+import random
+import time
+from typing import Iterable
+from typing import List
+
+import pytest
+
+import osparc_control
+from osparc_control.core import ControlInterface
+from osparc_control.errors import CommnadNotAcceptedError
+from osparc_control.errors import NoCommandReceivedArrivedError
+from osparc_control.models import CommandManifest
+from osparc_control.models import CommandParameter
+from osparc_control.models import CommnadType
+
+WAIT_FOR_DELIVERY = 0.1
+
+ALL_COMMAND_TYPES: List[CommnadType] = [
+    CommnadType.WITH_DELAYED_REPLY,
+    CommnadType.WITH_DELAYED_REPLY,
+    CommnadType.WITHOUT_REPLY,
+]
+
+# UTILS
+
+
+def _get_control_interface(
+    local_port: int, remote_port: int, exposed_interface: List[CommandManifest]
+) -> ControlInterface:
+    return ControlInterface(
+        remote_host="localhost",
+        exposed_interface=exposed_interface,
+        remote_port=remote_port,
+        listen_port=local_port,
+    )
+
+
+# FIXTURES
+
+
+@pytest.fixture
+def control_interface_a() -> Iterable[ControlInterface]:
+    control_interface = _get_control_interface(1234, 1235, [])
```

Comment:
if you do not use a concrete implementation such as Controller and Controlled or something similar, then it's not clear what these numbers are. tests are used to find how to use the library.

Reply:
I think this is a naming issue. I've done some renaming and this should be better now.